### PR TITLE
Moving SDL2, Sixense, and Faceshift to external projects

### PR DESCRIPTION
**Pull request by jherico**
This requires testing of the build to ensure that required DLLs are being copied.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a7f47408c702ad5e792d849840999e9ef92a8650 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 94eb5cc9de730a46b041df81762a51c5dd37a15c available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: daed8a9c88899bb07fc911f1c72fb0e1780f29d9 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88b7fea34d15dc31e068bca0fd2d1f65094c3d9a available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: af7d6dbee480d8cd29d8f489c9392228b69fab59 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

File: interface/resources/qml/ScrollingGraph2.qml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import QtQuick 2.1
```
Comment (ZappoMan) [inline interface/resources/qml/ScrollingGraph2.qml]: did you mean to add this?

File: cmake/externals/faceshift/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+# Hide this external target (for ide users)
+set_target_properties(${EXTERNAL_NAME} PROPERTIES FOLDER "hidden/externals")
+
+ExternalProject_Get_Property(${EXTERNAL_NAME} INSTALL_DIR)
+
+string(TOUPPER ${EXTERNAL_NAME} EXTERNAL_NAME_UPPER)
+set(${EXTERNAL_NAME_UPPER}_INCLUDE_DIRS ${INSTALL_DIR}/include CACHE FILEPATH "Path to gverb include directory")
+set(${EXTERNAL_NAME_UPPER}_LIBRARIES ${INSTALL_DIR}/lib/faceshift.lib CACHE FILEPATH "List of gverb libraries")
```
Comment (ZappoMan) [inline cmake/externals/faceshift/CMakeLists.txt]: message says gverb

File: cmake/externals/faceshift/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  LOG_BUILD 1
+)
+
+# Hide this external target (for ide users)
+set_target_properties(${EXTERNAL_NAME} PROPERTIES FOLDER "hidden/externals")
+
+ExternalProject_Get_Property(${EXTERNAL_NAME} INSTALL_DIR)
+
+string(TOUPPER ${EXTERNAL_NAME} EXTERNAL_NAME_UPPER)
+set(${EXTERNAL_NAME_UPPER}_INCLUDE_DIRS ${INSTALL_DIR}/include CACHE FILEPATH "Path to gverb include directory")
```
Comment (ZappoMan) [inline cmake/externals/faceshift/CMakeLists.txt]: message says gverb

**Comment by hifi-gustavo** [comment]
Build console output for commit: 920f632c211afd8be7eff7eeea49cfc127b80a60 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

File: interface/resources/qml/ScrollingGraph2.qml (hunk 1 above)
Comment (jherico) [inline interface/resources/qml/ScrollingGraph2.qml]: removed

File: cmake/externals/faceshift/CMakeLists.txt (hunk 2 above)
Comment (jherico) [inline cmake/externals/faceshift/CMakeLists.txt]: fixed

File: cmake/externals/faceshift/CMakeLists.txt (hunk 3 above)
Comment (jherico) [inline cmake/externals/faceshift/CMakeLists.txt]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: b8a1a83ae2e991378821db50414b7f0c80092c0d available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 418992dcb7d264c611c2ed5987ae03b1c946b853 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7a9f2252067c93058ff3d097e4fc81668990a42e available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9b0debac06cedce5477809f490d740cab29c9d4a available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9b0debac06cedce5477809f490d740cab29c9d4a available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5792895a42792e8514cc927f353483ac2be8266d available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a3ccde563fc08dcb49a31c1705157d86c5e07ee6 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: a3ccde563fc08dcb49a31c1705157d86c5e07ee6 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by ZappoMan** [comment]
verified: windows faceshift, sdl, and hydra all work.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5a6cf59a1f3d39d3246e507873a8d8f9df496945 available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8eb16d61db082d3b6ff6911b22ac9e7b6cdc46f available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8eb16d61db082d3b6ff6911b22ac9e7b6cdc46f available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8eb16d61db082d3b6ff6911b22ac9e7b6cdc46f available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: fcfab0fc513a96ae166318f613581cf083916e6e available at: https://gist.github.com/cb5e208392341f0ebf87<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6227.dmg

**Comment by thoys** [comment]
@ZappoMan not verified assignment-clients / domain-server only builds ;)

**Comment by thoys** [comment]
```
CMake Error at /usr/share/cmake-3.0/Modules/FindPackageHandleStandardArgs.cmake:136 (message):
  Could NOT find Sixense (missing: SIXENSE_INCLUDE_DIRS)
Call Stack (most recent call first):
  /usr/share/cmake-3.0/Modules/FindPackageHandleStandardArgs.cmake:343 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindSixense.cmake:22 (find_package_handle_standard_args)
  cmake/macros/TargetSixense.cmake:10 (find_package)
  libraries/input-plugins/CMakeLists.txt:15 (target_sixense)


-- Configuring incomplete, errors occurred!
See also "/home/tcars/hifi/build/CMakeFiles/CMakeOutput.log".
See also "/home/tcars/hifi/build/CMakeFiles/CMakeError.log".
Makefile:2042: recipe for target 'cmake_check_build_system' failed
make: *** [cmake_check_build_system] Error 1
```

**Comment by ZappoMan** [comment]
@thoys - did you clear your cmake cache?

**Comment by thoys** [comment]
doh silly me! works thanks @ZappoMan

**Comment by thoys** [comment]
I will add it to my auto-build shell script

**Pull request merged**